nix/hm-module.nix

```diff
@@ -114,11 +123,14 @@ in {
     home.sessionVariables =
       lib.mkIf cfg.recommendedEnvironment {NIXOS_OZONE_WL = "1";};
 
-    xdg.configFile."hypr/hyprland.conf" = lib.mkIf (cfg.extraConfig != null) {
+    xdg.configFile."hypr/hyprland.conf" = lib.mkIf (cfg.systemdIntegration || cfg.extraConfig != null || cfg.plugins != []) {
```

Comment:
I would think it would make sense to set an almost empty config if the user asks for it by setting those options. If we don't want to create the config when extraConfig is not set then there should be an error triggered when either option is set without extraConfig.

Reply:
Right, we could use asserts.

Reply:
I wont be able to make any changes for probably the next 6 hours at least if you want to add that. If not I'll do it when I can.

Reply:
I'm not particularly free either, so we'll see.

Reply:
Added a warning, but cfg.extraConfig is set to "" by default, causing it to be linked with home manager unless you explicitly opt out already.

Reply:
Alright, so will this break current users' configs?